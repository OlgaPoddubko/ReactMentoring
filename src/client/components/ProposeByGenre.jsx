import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { fetchRelatedMovies } from '../actions';

class ProposeByGenre extends Component {
  static propTypes = {
    currentMovie: PropTypes.object.isRequired,
  };

  constructor(props) {
    super(props);

    this.boundActions = bindActionCreators({fetchRelatedMovies}, this.props.dispatch);
  }

  render() {
    return (
      <div className="proposal">
        <div className="inner-proposal">
          Films by {this.props.currentMovie.genres ? this.props.currentMovie.genres[0] : ''} genre
        </div>
        <style jsx>{`
          .proposal {
            background: #f5f5f5;
          }
          .inner-proposal {
            padding: 15px 35px;
            max-width: 1130px;
            margin: 0 auto;
          }
        `}</style>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  currentMovie: state.gallery.currentMovie,
});

export default connect(mapStateToProps)(ProposeByGenre);
