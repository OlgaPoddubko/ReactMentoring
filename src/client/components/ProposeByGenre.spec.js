import React from 'react';
import { ProposeByGenre } from './ProposeByGenre';
import moviesArr from '../../../test/moviesArr';

describe('render ProposeByGenre', () => {
  it('should render ProposeByGenre', () => {
    const wrapper = shallow(<ProposeByGenre currentMovie = { moviesArr.data[0] }/>);
    expect(wrapper).toMatchSnapshot();
  });
});
