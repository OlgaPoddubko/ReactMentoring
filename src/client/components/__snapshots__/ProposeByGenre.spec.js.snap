// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ProposeByGenre should render ProposeByGenre 1`] = `
<div
  className="proposal"
>
  <div
    className="inner-proposal"
  >
    Films by 
    Romance
     genre
  </div>
  <style
    jsx={true}
  >
    
          .proposal {
            background: #f5f5f5;
          }
          .inner-proposal {
            padding: 15px 35px;
            max-width: 1130px;
            margin: 0 auto;
          }
        
  </style>
</div>
`;
